sycl/plugins/cuda/CMakeLists.txt

```diff
@@ -24,51 +23,19 @@ else()
   )
 endif()
 
-add_library(pi_cuda SHARED
-  "${sycl_inc_dir}/CL/sycl/detail/pi.h"
-  "${sycl_inc_dir}/CL/sycl/detail/pi.hpp"
-  "pi_cuda.hpp"
-  "pi_cuda.cpp"
-)
-
-add_dependencies(sycl-toolchain pi_cuda)
-
-set_target_properties(pi_cuda PROPERTIES LINKER_LANGUAGE CXX)
-
-target_include_directories(pi_cuda
-  PRIVATE
-    ${sycl_inc_dir}
-)
-
-target_link_libraries(pi_cuda
-  PRIVATE
+add_sycl_plugin(cuda
+  SOURCES
+    "${sycl_inc_dir}/CL/sycl/detail/pi.h"
+    "${sycl_inc_dir}/CL/sycl/detail/pi.hpp"
+    "pi_cuda.hpp"
+    "pi_cuda.cpp"
+  INCLUDE_DIRS
+    "${sycl_inc_dir}"
+  LIBRARIES
     OpenCL-Headers
     cudadrv
+  LINKER_SCRIPT "${CMAKE_CURRENT_SOURCE_DIR}/../ld-version-script.txt"
```

Comment:
It looks like all plug-ins use the same linker script. If so, can we move this setting inside `add_sycl_plugin` function to avoid duplication?

The same comment for `INCLUDE_DIRS` - all plug-ins add "${sycl_inc_dir}", so let's move it inside `add_sycl_plugin`.

Reply:
Done